raphtory/src/db/api/view/serialise.rs

```diff
@@ -0,0 +1,1092 @@
+                let layer_id = *ee.edge.layer().expect("exploded layers");
+
+                for (prop_name, prop) in ee.properties().constant() {
+                    let key = self
+                        .edge_meta()
+                        .const_prop_meta()
+                        .get_id(&prop_name)
+                        .unwrap();
+                    const_edges_props.push(serialise::UpdateEdgeConstProps {
+                        src,
+                        dst,
+                        layer_id: layer_id as u64,
+                        properties: Some(as_prop_pair(key as u64, &prop)?),
+                    });
+                }
+
+                for ee in ee.explode() {
+                    edges.push(AddEdge {
+                        src,
+                        dst,
+                        properties: None,
+                        time: ee.time().expect("exploded edge"),
+                        layer_id: Some(layer_id as u64),
+                    });
+
+                    for (prop_name, prop_view) in ee.properties().temporal() {
+                        for (time, prop) in prop_view.iter() {
+                            let key = self
+                                .edge_meta()
+                                .temporal_prop_meta()
+                                .get_id(&prop_name)
+                                .unwrap();
+                            edges.push(AddEdge {
+                                src,
+                                dst,
+                                properties: Some(as_prop_pair(key as u64, &prop)?),
+                                time,
+                                layer_id: Some(layer_id as u64),
+                            });
+                        }
+                    }
+
+                    for time in ee.deletions() {
+                        del_edges.push(DelEdge {
+                            src,
+                            dst,
+                            time,
+                            layer_id: Some(layer_id as u64),
+                        });
+                    }
+                }
+            }
+        }
+
+        encode_length_delimiter(edges.len(), &mut bytes)?;
+        for edge in edges {
+            edge.encode_length_delimited(&mut bytes)?;
+        }
+
+        encode_length_delimiter(del_edges.len(), &mut bytes)?;
+        for del_edge in del_edges {
+            del_edge.encode_length_delimited(&mut bytes)?;
+        }
+
+        encode_length_delimiter(const_edges_props.len(), &mut bytes)?;
+        for const_edge_props in const_edges_props {
+            const_edge_props.encode_length_delimited(&mut bytes)?;
+        }
+
+        Ok(bytes)
+    }
+}
+
+impl<
+        'graph,
+        G: InternalAdditionOps
+            + GraphViewOps<'graph>
+            + DelegatePropertyAdditionOps
+            + InternalDeletionOps,
+    > StableDecode for G
+{
+    fn decode_from_bytes(buf: &[u8], graph: &Self) -> Result<(), GraphError> {
```

Comment:
Probably would be better to make a new graph here (you can use an `InternalGraph` here and then add `decode_from_bytes` constructors to `Graph` and `PersistentGraph` which call this function and wrap the `InternalGraph`)

Reply:
I wanted this to apply to all graphs not just Graph and PersistentGraph